pgxn/neon/neon.c

```diff
@@ -85,3 +93,129 @@ backpressure_throttling_time(PG_FUNCTION_ARGS)
 {
 	PG_RETURN_UINT64(BackpressureThrottlingTime());
 }
+
+
+#define PG_LOG_DIR 			"log"
+#define POSTGRES_LOG 		"postgres_log"
+#define LOG_TABLE_N_COLUMS	26
+
+typedef struct {
+	char*  path;
+	time_t ctime;
+} LogFile;
+
+typedef struct
+{
+	Relation    log_table;
+	List*       log_files;
+	CopyFromState copy_state;
+	ListCell*   curr_log;
+} LogfileContext;
+
+static int cmp_log_ctime(const ListCell *a, const ListCell *b)
+{
+	LogFile* la = (LogFile*)lfirst(a);
+	LogFile* lb = (LogFile*)lfirst(b);
+	return la->ctime < lb->ctime ? -1 : la->ctime == lb->ctime ? 0 : 1;
+}
+
+
+
+Datum
+read_postgres_log(PG_FUNCTION_ARGS)
```

Comment:
Was there any reason to not use `pg_read_file()` or `read_text_file()` ?
If so, please add the comment.